# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1294917633 18000
#      Thu Jan 13 06:20:33 2011 -0500
# Node ID fc35194006d642920b185b47c0947745b67a6df0
# Parent  41866901ec624bd6967a3c218e685af6c9e29fdc
oct-fftw.cc: more constructor tweaks

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,15 @@
+2011-01-13  John W. Eaton  <jwe@octave.org>
+
+	* oct-fftw.cc (octave_fftw_planner::octave_fftw_planner,
+	octave_float_fftw_planner::octave_float_fftw_planner):
+	Avoid duplicate initializations.  Correctly initialize RN to
+	empty dim_vector.
+
 2011-01-13  John W. Eaton  <jwe@octave.org>
 
 	* Sparse.cc (Sparse::Sparse): Initialize all data members in
 	initialization lists.
 
 2011-01-13  John W. Eaton  <jwe@octave.org>
 
 	* Array.cc (rec_permute_helper::rec_permute_helper,
diff --git a/liboctave/oct-fftw.cc b/liboctave/oct-fftw.cc
--- a/liboctave/oct-fftw.cc
+++ b/liboctave/oct-fftw.cc
@@ -51,31 +51,24 @@ octave_fftw_planner *octave_fftw_planner
 // FIXME -- if we can ensure 16 byte alignment in Array<T>
 // (<T> *data) the FFTW3 can use SIMD instructions for further
 // acceleration.
 
 // Note that it is profitable to store the FFTW3 plans, for small
 // ffts.
 
 octave_fftw_planner::octave_fftw_planner (void)
-  : meth (), rplan (), rd (0), rs (0), rr (0), rh (0), rn (0),
+  : meth (ESTIMATE), rplan (0), rd (0), rs (0), rr (0), rh (0), rn (),
     rsimd_align (false)
 {
-  meth = ESTIMATE;
-
   plan[0] = plan[1] = 0;
   d[0] = d[1] = s[0] = s[1] = r[0] = r[1] = h[0] = h[1] = 0;
   simd_align[0] = simd_align[1] = false;
   inplace[0] = inplace[1] = false;
   n[0] = n[1] = dim_vector ();
-
-  rplan = 0;
-  rd = rs = rr = rh = 0;
-  rsimd_align = false;
-  rn = dim_vector ();
   
   // If we have a system wide wisdom file, import it.
   fftw_import_system_wisdom ();
 }
 
 bool
 octave_fftw_planner::instance_ok (void)
 {
@@ -366,32 +359,25 @@ octave_fftw_planner::do_method (FftwMeth
   else
     ret = UNKNOWN;
   return ret;
 }
 
 octave_float_fftw_planner *octave_float_fftw_planner::instance = 0;
 
 octave_float_fftw_planner::octave_float_fftw_planner (void)
-  : meth (), rplan (), rd (0), rs (0), rr (0), rh (0), rn (0),
+  : meth (ESTIMATE), rplan (0), rd (0), rs (0), rr (0), rh (0), rn (),
     rsimd_align (false)
 {
-  meth = ESTIMATE;
-
   plan[0] = plan[1] = 0;
   d[0] = d[1] = s[0] = s[1] = r[0] = r[1] = h[0] = h[1] = 0;
   simd_align[0] = simd_align[1] = false;
   inplace[0] = inplace[1] = false;
   n[0] = n[1] = dim_vector ();
-
-  rplan = 0;
-  rd = rs = rr = rh = 0;
-  rsimd_align = false;
-  rn = dim_vector ();
-  
+ 
   // If we have a system wide wisdom file, import it.
   fftwf_import_system_wisdom ();
 }
 
 bool
 octave_float_fftw_planner::instance_ok (void)
 {
   bool retval = true;
